packages/atomic/src/components/search/result-template-components/atomic-result-localized-text/atomic-result-localized-text.ts

```diff
@@ -0,0 +1,81 @@
+import {isNullOrUndefined} from '@coveo/bueno';
+import {Result, ResultTemplatesHelpers} from '@coveo/headless';
+import {Component, Prop, State} from '@stencil/core';
+import {
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../../utils/initialization-utils';
+import {MapProp} from '../../../../utils/props-utils';
+import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
+import {ResultContext} from '../result-template-decorators';
+
+/**
+ * The `atomic-result-localized-text` component renders a target i18n localized string using field values as parameter.
+ *
+ * Given this i18n configuration:
+ * ```
+ * searchInterface.i18n.addResourceBundle('en', 'translation', {
+ *    foo: 'Hello world {{a_field_value}} !',
+ * });
+ * ```
+ *
+ * The component could be configured in such a way to replace `{{a_field_value}}` with a dynamic field value from the result:
+ * ```
+ * <atomic-result-localized-text key="foo" field-value-myfield="a_field_value"></atomic-result-localized-text>
+ * ```
+ *
+ * @MapProp name: fieldValue;attr: field-value;docs: The field and values that define which result items to extract as an i18n parameter. For example, the following configuration extract the value of `filetype` from a result, and assign it to the i18n parameter `the_parameter`: `field-value-filetype="the_parameter"`;type: Record<string, string> ;default: {}
```

Comment:
Since `fieldValue` seems to be talking about the value of a field, I think it may be clearer for users if we called this parameter something like: `fieldAndVariable` or `fieldAndTarget`.
```suggestion
 * @MapProp name: fieldAndVariable;attr: field-and-variable;docs: The field from which to extract the target string and the variable used to map it to the target i18n parameter. For example, the following configuration extracts the value of `author` from a result, and assign it to the i18n parameter `name`: `field-value-author="name"`;type: Record<string, string> ;default: {}
```